manifest/v1alpha/service/validation_test.go

```diff
@@ -100,6 +100,378 @@ func TestValidate_Spec(t *testing.T) {
+			Code: rules.ErrorCodeRequired,
+		})
+	})
+
+	t.Run("invalid timeZone", func(t *testing.T) {
+		svc := validService()
+		svc.Spec.ReviewCycle = &ReviewCycle{
+			StartTime: "2023-01-01T00:00:00",
+			TimeZone:  "Invalid/Timezone",
+			RRule:     "FREQ=MONTHLY;INTERVAL=1",
+		}
+		err := validate(svc)
+		testutils.AssertContainsErrors(t, svc, err, 1, testutils.ExpectedError{
+			Prop: "spec.reviewCycle.timeZone",
+			Code: "string_time_zone",
+		})
+	})
+
+	t.Run("empty rrule", func(t *testing.T) {
+		svc := validService()
+		svc.Spec.ReviewCycle = &ReviewCycle{
+			StartTime: "2023-01-01T00:00:00",
+			TimeZone:  "UTC",
+			RRule:     "",
+		}
+		err := validate(svc)
+		testutils.AssertContainsErrors(t, svc, err, 1, testutils.ExpectedError{
+			Prop: "spec.reviewCycle.rrule",
+			Code: rules.ErrorCodeRequired,
+		})
+	})
+
+	t.Run("rrule frequency validation", func(t *testing.T) {
+		tests := []struct {
+			name        string
+			rrule       string
+			expectError bool
+			expectedMsg string
+		}{
+			{
+				name:        "missing FREQ",
+				rrule:       "INTERVAL=1",
+				expectError: true,
+				expectedMsg: "RRULE property FREQ is required",
+			},
+			{
+				name:        "hourly frequency not allowed",
+				rrule:       "FREQ=HOURLY;INTERVAL=1",
+				expectError: true,
+				expectedMsg: "frequency must be DAILY, WEEKLY, MONTHLY, or YEARLY",
+			},
+			{
+				name:        "minutely frequency not allowed",
+				rrule:       "FREQ=MINUTELY;INTERVAL=60",
+				expectError: true,
+				expectedMsg: "frequency must be DAILY, WEEKLY, MONTHLY, or YEARLY",
+			},
+			{
+				name:        "daily frequency allowed",
+				rrule:       "FREQ=DAILY;INTERVAL=1",
+				expectError: false,
+			},
+			{
+				name:        "weekly frequency allowed",
+				rrule:       "FREQ=WEEKLY;INTERVAL=1",
+				expectError: false,
+			},
+			{
+				name:        "monthly frequency allowed",
+				rrule:       "FREQ=MONTHLY;INTERVAL=1",
+				expectError: false,
+			},
+			{
+				name:        "yearly frequency allowed",
+				rrule:       "FREQ=YEARLY;INTERVAL=1",
+				expectError: false,
+			},
+			{
+				name:        "single occurrence with any frequency not allowed",
+				rrule:       "FREQ=HOURLY;COUNT=1",
+				expectError: true,
+				expectedMsg: "rrule must have continuous occurrences",
+			},
+			{
+				name:        "single occurrence with daily frequency not allowed",
+				rrule:       "FREQ=DAILY;COUNT=1",
+				expectError: true,
+				expectedMsg: "rrule must have continuous occurrences",
+			},
+			{
+				name:        "multiple limited occurrences not allowed",
+				rrule:       "FREQ=WEEKLY;COUNT=5",
+				expectError: true,
+				expectedMsg: "rrule must have continuous occurrences",
+			},
+			{
+				name:        "zero interval not allowed",
+				rrule:       "FREQ=DAILY;INTERVAL=0",
+				expectError: false, // rrule library accepts 0 and converts to 1
+			},
+			{
+				name:        "negative interval not allowed",
+				rrule:       "FREQ=WEEKLY;INTERVAL=-1",
```

Comment:
damn.. itsn't this already validated by the library?!

Reply:
it' defaults to 1, but during testing it wasn't clear that it would default to 1, so I've added seperate check.